app/src/main/kotlin/com/intive/picover/parties/view/PartiesScreen.kt

```diff
@@ -26,42 +25,33 @@ import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
-import androidx.navigation.NavHostController
+import cafe.adriel.voyager.core.screen.Screen
+import cafe.adriel.voyager.hilt.getScreenModel
 import com.intive.picover.R
 import com.intive.picover.common.error.PicoverGenericError
 import com.intive.picover.common.loader.PicoverLoader
 import com.intive.picover.common.viewmodel.state.MVIStateType
+import com.intive.picover.main.navigation.NavControllerHolder
 import com.intive.picover.main.theme.Typography
-import com.intive.picover.parties.model.PartiesEvent
-import com.intive.picover.parties.model.PartiesSideEffect.NavigateToAddParty
-import com.intive.picover.parties.model.PartiesSideEffect.NavigateToPartyDetails
 import com.intive.picover.parties.model.Party
 import com.intive.picover.parties.viewmodel.PartiesViewModel
-import kotlinx.coroutines.flow.collectLatest
 
-@Composable
-fun PartiesScreen(
-	viewModel: PartiesViewModel,
-	navController: NavHostController,
-) {
-	val state by viewModel.state.collectAsState()
-	LaunchedEffect(true) {
-		viewModel.sideEffects.collectLatest { effect ->
-			when (effect) {
-				is NavigateToPartyDetails -> navController.navigate("partyDetails/${effect.partyId}")
-				is NavigateToAddParty -> navController.navigate("parties/addParty")
-			}
-		}
-	}
-	when (state.type) {
-		MVIStateType.LOADING -> PicoverLoader(modifier = Modifier.fillMaxSize())
-		MVIStateType.LOADED -> LoadedContent(
-			parties = state.parties,
-			onPartyClick = { viewModel.effect(NavigateToPartyDetails(it)) },
-			onFabClick = { viewModel.effect(NavigateToAddParty) },
-		)
+class PartiesScreen : Screen {
+
+	@Composable
+	override fun Content() {
+		val viewModel = getScreenModel<PartiesViewModel>()
+		val state by viewModel.state.collectAsState()
+		when (state.type) {
+			MVIStateType.LOADING -> PicoverLoader(modifier = Modifier.fillMaxSize())
+			MVIStateType.LOADED -> LoadedContent(
+				parties = state.parties,
+				onPartyClick = { NavControllerHolder.navigate("partyDetails/$it") },
+				onFabClick = { NavControllerHolder.navigate("parties/addParty") },
```

Comment:
Once everything is migrated we will figure out how to handle navigation properly and how to test it

Reply:
Maybe we can `navController` property public and use static import for `NavControllerHolder`? Since it will be removed anyway we can reduce number of changed lines in the end because, e.g. `navController.navigate("parties/addParty")` would stay as it is.

Reply:
Actually non-null assertion `!!` would be still needed,  so I guess amount of changes would be similar, I would just keep it as it is for now